add estimate_cost() method to Job and Batch

tylerflex opened the pull request:
I noticed that the `estimate_cost()` was looking for "flex_unit" in the task info dict, which was None. I changed it to `estFlexUnit` to fix things.

Note: it seems that when the estimated cost is not available, it gets returned as 0, which is potentially confusing. Should we return `None` in this case?

another note: if the job has not started, it seems like the estimated credits are never available. Is that supposed to be the way it works?

momchil-flex reviewed (commented):
> I noticed that the estimate_cost() was looking for "flex_unit" in the task info dict, which was None. I changed it to estFlexUnit to fix things.

Do you mean fix things in the tests?

> Note: it seems that when the estimated cost is not available, it gets returned as 0, which is potentially confusing. Should we return None in this case?

I think this is why it was not `estFlexUnit`. I'm a bit confused as to why `flex_unit` works though, I need to disentangle what's happening as I don't remember.

> another note: if the job has not started, it seems like the estimated credits are never available. Is that supposed to be the way it works?

I think this may be just because you were running latest develop and the metadata needs a rebuild to work for that? I tried v2.2.0 and estimate_cost works before a task is started.

momchil-flex commented:
Yeah so `flex_unit` from the metadata response was correct, it should *not* be `estFlexUnit` from the task info since like you say this is 0 initially and only populated when the task is running. Unfortunately there are many overlapping variables, some of them were kept for backwards compatibility, but it would be great if we can clean things up a bit @lei-flex 

@tylerflex you were most likely confused because the metadata api was not working for the latest state of the code.

momchil-flex commented:
I have updated the version to `2.3.0rc2` and built a corresponding solver and metadata webapi.

tylerflex commented:
ok I'm a bit confused then. So the web.monitor uses "estFlexUnit" (because it knows the task is running) but everything else should use "flex_unit"? And yea before I noticed that `estimate_cost(task_id)` was just not working, period. And the warning message was something like "Couldn't compute SimulationData", which was confusing.

Not sure what to do here.. I guess revert my last commit?

tylerflex commented on CHANGELOG.md:
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Causality validation in `PoleResidue` model.
 - Group index calculation added to mode monitors and available through `ModeData.n_group`.
 - Support for `JaxGeometryGroup` in adjoint plugin.
+- a `estimate_cost()` method for `Job` and `Batch`.
```
Thanks, fixed

tylerflex commented:
I dropped the last commit and fixed the changelog typo

momchil-flex commented:
Looks good, only there seem to be two lines in the changelog now (maybe because of the merge strategy and something happening during rebase). You can merge this after you fix this, and I'll be working on estimate_cost these days anyway.

> ok I'm a bit confused then. So the web.monitor uses "estFlexUnit" (because it knows the task is running) but everything else should use "flex_unit"? And yea before I noticed that estimate_cost(task_id) was just not working, period. And the warning message was something like "Couldn't compute SimulationData", which was confusing.

Yeah it's definitely confusing. I remember now: `flex_unit` is returned by the metadata api if it succeeds. `estFlexUnit` is a task field that's updated only when the simulation is started, based on the `flex_unit` from the metadata web api, or a metadata pipeline run if the web api fails. Basically, before you run a simulation, you can keep changing the uploaded json (this is much more common for GUI), and keep calling the metadata webapi to get different `flex_unit`. The `estFlexUnit` is frozen once the simulation is started and is used for the actual billing at the end.

tylerflex commented:
Ok I 
1. fixed the changelog
2. added a estimate_cost() call and console.log so that the Batch now prints a message about the total estimated cost (similar to a single task).

<img width="960" alt="image" src="https://github.com/flexcompute/tidy3d/assets/92756888/f14575a2-831f-4932-be8a-3363739bf644">

Should be all set on this I think.
